server.js

```diff
@@ -1,10 +1,106 @@
-const express = require('express'); // remember to install your npm packages
-
+const mongoose = require('mongoose');
+const express = require('express');
+const bodyParser = require('body-parser');
+mongoose.connect('mongodb://localhost/budgetDB');
+const Budget = require('./Models/budget.js');
+const Category = require('./Models/category.js');
+const Expense = require('./Models/expense.js');
 const server = express();
+const to = require('./to')
+
+server.use(bodyParser.json());
 
-// add your server code
+server.post('/budget', async (req, res) => {
+    const bud = req.body;
+    if (bud.title && bud.budgetAmount) {
+        const budget = new Budget(bud);
+        const [err] = await to(budget.save());
+        if (err) {
+            res.status(500).json({
+                error: 'budgetAmount or title is missing.'
+            })
+        } else {
+            res.status(200).json({
+                sucess: 'Budget got saved.'
+            });
+        }
+    } else {
+        res.status(422).json({
+            error: 'budgetAmount or title is missing.'
+        })
+    }
+});
+server.post('/category', (req, res) => {
+    const title = req.body;
+    if (title) {
+        const category = new Category(title);
+        category.save();
+        res.status(200).json({
+            sucess: 'Category got saved.'
+        });
+    } else {
+        res.status(422).json({
+            error: 'Title is missing.'
+        })
+    }
+});
+//Trying a try catch
+server.get('/category', async (req, res) => {
+    try {
+        const cat = await Category.find({});
+        res.status(200).json(cat);
+    } catch (err) {
+        res.status(422).json(err);
+    }
+});
+//I am going to get  the budget and category id's from the client.
 
+server.get('/expense', async (req, res) => {
+    const [err, exp] = await to(Expense.find().populate('budget category'));
+    if (err) res.status(err).json({
+        error: 'Expense not found '
+    })
+    if (exp) res.status(200).json(exp);
+});
+server.post('/expense', async (req, res) => {
+    const exp = req.body;
+    const [err_b, b] = await to(Budget.findOne({
```

Comment:
the results of evaluating line 67 will invariably always be [null, null] and hence the error message on line 74 will always be invoked.